Design.md

```diff
@@ -62,9 +62,9 @@ Library routines may allocate memory internally, but not in a way that affects t
 Exceptions are inefficient and use the GC. Of course, they cannot be used in `nothrow` code. Examine each use of an Exception to see if it can be designed out of existence, like the Replacement Character method above. Design the return value such that an error is not necessary - for example, a string search function can return an empty string if not found rather than throw an Exception.
 Investigate the use of Option types for error returns.
 
-### Header Only
+### Source Only
 
-By making the library header only, it becomes inured against variations in compiler flags. Whether it's a static or dynamic library becomes irrelevant, and there won't be impedance mismatches.
+Currently, Phobos is distributed as a separately compiled library with "header" files that contain on the necessary Template implementations. The net result is that Phobos is primarily a source-only library in practice. By formalizing the library as a source only library, it becomes inured against variations in compiler flags. Whether it's a static or dynamic library becomes irrelevant, and there won't be impedance mismatches. DRuntime will remain a separately compiled library.
```

Comment:
s/on/only/. 

Is it true that we essentially ship .di files? I have the full source code in `/usr/include`.

Reply:
We ship the binary compiled phobos library, along with full source (not di files).

Reply:
My understanding is that we compile against the full source, but that there is _also_ a compiled library that ships underneath it with the entire library of compiled symbols for Phobos. As best I can tell, we _could_ ship DI files and it would still work, but we don't. The reasons why it is this way, and the mechanics behind which (source or library) is actually getting used, still elude me.